torch/optim/adam.py

```diff
@@ -556,26 +564,13 @@ def _multi_tensor_adam(
     if len(params) == 0:
         return
 
-    if isinstance(lr, Tensor) and not capturable:
-        raise RuntimeError(
-            "lr as a Tensor is not supported for capturable=False and foreach=True"
-        )
-
-    if isinstance(beta1, Tensor):
```

Comment:
looks like the beta1 branch was accidentally deleted? Or does it work now?

Reply:
`Adam.__init__` already has the same check
https://github.com/pytorch/pytorch/blob/4233a779603207f19033cd433d2961c93b932cb4/torch/optim/adam.py#L72-L85

I thought it is redundant.

Reply:
Ah it's not redundant because people could access _multi_tensor_adam through the functional API (not only through the constructor)

Reply:
I reverted it: https://github.com/pytorch/pytorch/pull/145674/commits/32812c31390ebd29af156e94f467421f8ac92241